src/main/java/org/apache/sysml/hops/NaryOp.java

```diff
@@ -159,7 +159,8 @@ else if ( areDimsBelowThreshold() )
 		setRequiresRecompileIfNecessary();
 		
 		//ensure cp exec type for single-node operations
-		if ( _op == OpOpN.PRINTF )
+		//TODO eval function only supports CP execution type
```

Comment:
please remove the TODO as eval will be be similar to all other function calls always be executed in CP.